modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2202,12 +2202,44 @@ def map(
         PandasDataframe
             A new dataframe.
         """
-        map_fn = (
-            self._partition_mgr_cls.lazy_map_partitions
-            if lazy
-            else self._partition_mgr_cls.map_partitions
-        )
-        new_partitions = map_fn(self._partitions, func, func_args, func_kwargs)
+        if self.num_parts <= 1.5 * CpuCount.get():
+            # block-wise map
+            map_fn = (
+                self._partition_mgr_cls.lazy_map_partitions
+                if lazy
+                else self._partition_mgr_cls.map_partitions
+            )
+            new_partitions = map_fn(self._partitions, func, func_args, func_kwargs)
+        else:
+            # axis-wise map
+            # we choose an axis for a combination of partitions whose size is closer to the number of CPUs
+            axis = 0
+            if abs(self._partitions.shape[0] - CpuCount.get()) < abs(
+                self._partitions.shape[1] - CpuCount.get()
+            ):
+                axis = 1
+
+            column_splits = (
+                self._partitions.shape[0] * self._partitions.shape[1]
+            ) // CpuCount.get()
+
+            if axis == 1 or column_splits <= 1:
+                # splitting by full axis partitions
+                new_partitions = self._partition_mgr_cls.map_axis_partitions(
+                    axis,
+                    self._partitions,
+                    func,
+                    keep_partitioning=True,
+                    map_func_args=func_args,
+                    **(func_kwargs if func_kwargs is not None else {}),
+                )
```

Comment:
?
```suggestion
                def new_map_func(*args):
                    return func(*args, **func_kwargs)
                new_partitions = self._partition_mgr_cls.map_axis_partitions(
                    axis,
                    self._partitions,
                    new_map_func,
                    keep_partitioning=True,
                    map_func_args=func_args,
                )
```

Reply:
Thanks for your suggestion. I used it, but move this function into lambda.